diff -Nur redhat-rpm-config-9.1.0.orig/macros redhat-rpm-config-9.1.0/macros
--- redhat-rpm-config-9.1.0.orig/macros	2013-07-24 14:45:29.972653928 -0600
+++ redhat-rpm-config-9.1.0/macros	2013-07-24 14:46:14.419922188 -0600
@@ -18,6 +18,8 @@
 %_mandir		%{_prefix}/share/man
 
 %_defaultdocdir		%{_prefix}/share/doc
+%_pkgdocdir             %{_docdir}/%{name}
+%_docdir_fmt            %%{NAME}
 
 %_fmoddir		%{_libdir}/gfortran/modules
 
