diff -up redhat-rpm-config-9.1.0/macros.record-switches redhat-rpm-config-9.1.0/macros
--- redhat-rpm-config-9.1.0/macros.record-switches	2013-04-22 13:17:29.319358036 +0300
+++ redhat-rpm-config-9.1.0/macros	2013-04-22 13:17:44.665308442 +0300
@@ -171,7 +171,7 @@ package or when debugging this package.\
 %_hardened_cflags	%{?_hardened_build:%{_hardening_cflags}}
 %_hardened_ldflags	%{?_hardened_build:%{_hardening_ldflags}}
 
-%__global_cflags	-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 %{_hardened_cflags}
+%__global_cflags	-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -grecord-gcc-switches %{_hardened_cflags}
 %__global_ldflags	-Wl,-z,relro %{_hardened_ldflags}
 
 # Use these macros to differentiate between RH and other KMP implementation(s).
