--- redhat-rpm-config-8.0.45/macros.java	2006-09-05 16:24:06.000000000 -0400
+++ redhat-rpm-config-8.0.45/macros	2006-09-05 16:24:24.000000000 -0400
@@ -86,6 +86,7 @@
     /usr/lib/rpm/redhat/brp-strip-static-archive %{__strip} \
     /usr/lib/rpm/redhat/brp-strip-comment-note %{__strip} %{__objdump} \
     /usr/lib/rpm/brp-python-bytecompile \
+    /usr/lib/rpm/redhat/brp-java-repack-jars \
 %{nil}
 
 #    /usr/lib/rpm/redhat/brp-implant-ident-static
