diff -urN redhat-rpm-config-8.0.45_orig/find-requires redhat-rpm-config-8.0.45/find-requires
--- redhat-rpm-config-8.0.45_orig/find-requires	2006-07-30 19:58:01.000000000 +0100
+++ redhat-rpm-config-8.0.45/find-requires	2006-10-12 04:55:41.000000000 +0100
@@ -5,11 +5,6 @@
 # sonames, script interpreters, and perl modules.
 #
 
-if [ "$1" ]
-then
-   package_name = "$1"
-fi
-
 ulimit -c 0
 
 #
@@ -144,21 +139,18 @@
 # cheat a little here by looking first for a kernel, then for a kmod.
 #
 
-is_kmod=1
+unset is_kmod
+
 for f in $filelist; do
     if [ $(echo "$f" | sed -r -ne 's:^.*/lib/modules/(.*)/(.*).ko$:\2:p') ]
     then
-        is_kernel=1;
-    fi
-    if [ $(echo "$f" | sed -r -ne 's:^.*/boot/(.*):\1:p') ]
+        is_kmod=1;
+    elif [ $(echo "$f" | sed -r -ne 's:^.*/boot/(.*):\1:p') ]
     then
-        unset is_kmod;
+	unset is_kmod;
+	break;
     fi
 done
-if [ ! "$is_kernel" ] || [ "$package_name" == "kernel" ]
-then
-    unset is_kmod
-fi
 
 [ -x /usr/lib/rpm/redhat/find-requires.ksyms ] && [ "$is_kmod" ] &&
     printf "%s\n" "${filelist[@]}" | /usr/lib/rpm/redhat/find-requires.ksyms
