diff -up redhat-rpm-config-9.1.0/macros.configfoo redhat-rpm-config-9.1.0/macros
--- redhat-rpm-config-9.1.0/macros.configfoo	2013-04-22 13:42:25.984613491 +0300
+++ redhat-rpm-config-9.1.0/macros	2013-04-22 13:43:27.053420367 +0300
@@ -33,6 +33,9 @@
   CXXFLAGS="${CXXFLAGS:-%optflags}" ; export CXXFLAGS ; \
   FFLAGS="${FFLAGS:-%optflags -I%_fmoddir}" ; export FFLAGS ; \
   LDFLAGS="${LDFLAGS:-%__global_ldflags}"; export LDFLAGS; \
+  for i in $(find . -name config.guess -o -name config.sub) ; do \
+      [ -f /usr/lib/rpm/redhat/$(basename $i) ] && %{__rm} -f $i && %{__cp} -fv /usr/lib/rpm/redhat/$(basename $i) $i ; \
+  done ; \
   %{_configure} --build=%{_build} --host=%{_host} \\\
 	--program-prefix=%{?_program_prefix} \\\
 	--disable-dependency-tracking \\\
